api/e2e/test/01_create_router_test.go

```diff
@@ -81,40 +81,62 @@ func TestCreateRouter(t *testing.T) {
 					assert.Equal(t, http.StatusOK, response.StatusCode,
 						"Unexpected response (code %d): %s",
 						response.StatusCode, string(responsePayload))
-					actualResponse := gjson.GetBytes(responsePayload, "#.data.response").String()
```

Comment:
Somehow when you rewrote the entire expected and actual responses explicitly, the tests seem to work perfectly fine but I'm still unsure of the original cause of the non-deterministic test outcomes initially. One possible (but still unlikely I feel) cause might've been the way the responses were extracted from the original JSON body with the `gjson` package in that the response order might not be preserved, but I can't confirm this 👀

Reply:
Yes, may assumptions are along these lines too